File: Core/Entities/Category.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities;

public class Category : BaseEntity
{
    public string CategoryName { get; set; } = string.Empty;
    public string? Icon { get; set; }
    public Guid? ParentCategoryId { get; set; }

    // Navigation Properties
    public virtual Category? ParentCategory { get; set; }

    public virtual ICollection<Category> SubCategories { get; set; } = new List<Category>();
    public virtual ICollection<Product> Products { get; set; } = new List<Product>();

}

File: web/src/app/shared/models/category.ts
export type Category = {
  id: string;
  categoryName: string;
  icon?: string;
  parentCategoryId?: string;
  productCount: number;
  createdAt: string;
  updatedAt: string;
  subCategories: Category[];
}

export type AddCategory ={
  categoryName: string;
  icon?: string;
  parentCategoryId?: string;
}
